packages/chain-adapters/src/utxo/UtxoBaseAdapter.ts

```diff
@@ -434,10 +453,7 @@ export abstract class UtxoBaseAdapter<T extends UtxoChainId> implements IChainAd
     receiverAddress,
     signTxInput,
   }: SignAndBroadcastTransactionInput<T>): Promise<string> {
-    await Promise.all([
-      validateAddress(senderAddress),
-      receiverAddress !== CONTRACT_INTERACTION && validateAddress(receiverAddress),
-    ])
+    await (receiverAddress !== CONTRACT_INTERACTION && validateAddress(receiverAddress))
```

Comment:
Keeps only the receive side check now - the naming is odd but this refers (or at least *should* refer to) the *final* recipient e.g for a cross-chain swap, possibly a receive address on a diff chain